torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2543,7 +2543,11 @@ XLATensor XLATensor::transpose(const XLATensor& input, xla::int64 dim0,
 }
 
 void XLATensor::transpose_(XLATensor& input, xla::int64 dim0, xla::int64 dim1) {
-  input.SetIrValue(ir::ops::TransposeOp(input.GetIrValue(), dim0, dim1));
+  auto input_shape = input.shape();
+  auto permute_dims = XlaHelpers::MakeTransposePermutation(
```

Comment:
Thanks @bdhirsh for the fix! It's better that if we can wrap these logic into a transpose_ op in `ir::ops` I think. :D

Reply:
I think it is OK to leave it like this. The original purpose of `TransposeOp` was to share the same implementation between `transpose` and `transpose_` but `transpose` now no longer uses `TransposeOp` due to similar view problems. I think we should remove TransposeOp in this pr too since it is not needed.

Reply:
Yeah, I guess all that `TransposeOp` does it generate a `Permute` node. And `XLATensor::transpose_()` will end up doing the same thing by adding the view_info to the current tensor's view:
```
ViewInfo view_info(ViewInfo::Type::kPermute, input_shape, permute_dims);
return input.ModifyCurrentView(std::move(view_info));
```
Since that'll append a `kPermute` view_info to the view, and materializing the view will eventually create a `Permute` node here: https://github.com/pytorch/xla/blob/86998fe3694da1639954f7d0628557ca7709b9e7/torch_xla/csrc/view.cpp#L40